### PR TITLE
Use Poetry for dependencies

### LINKED ISSUE #255: Implement Python Poetry
### Feature request

Let's use modern Python tooling

### Motivation

Let's use modern Python tooling

### Other

_No response_

### PR DESCRIPTION
**Pull request by 0dm**
**What kind of change does this PR introduce?**

**Checklist**
* [x] My code follows the style guidelines of this project
* [x] I have pefomed a self-review of my code
* [x] If applicable, I have added tests to prove my fix is functional/effective
* [x] I have linted my code locally prior to submission
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation (e.g. README.md, requirements.txt)
* [x] New and existing unit tests pass locally with my changes

**Summary**

Uses [poetry](https://python-poetry.org).

I have copied all requirements to pyproject.toml. Documentation changes are still needed. This should be working AFAIK, but please test it as well.

- resolves #255 

**How can your code be run and tested?**

usage: `poetry shell` / `poetry install` / `poetry run pytest`

### CODE REVIEW

File: poetry.lock
Comment (0dm) [inline poetry.lock:1]: While this is a large file, the developers recommend that we commit the .lock file (https://python-poetry.org/docs/basic-usage/#commit-your-poetrylock-file-to-version-control).

File: openadapt/record.py [hunk 1]
```diff
@@ -642,5 +642,10 @@ def record(
     logger.info(f"saved {recording_timestamp=}")
 
 
+# entry point
+def start():
```
Comment (abrichr) [inline openadapt/record.py:646]: What is this function for?

Comment (0dm) [reply openadapt/record.py:646]: we need a function that acts an entry point / main() for us to use `poetry run`, called like `openadapt.record:start`. Please let me know if there's a better way to do this, I am aware there's a guard for this already but openadapt.record cannot be called through poetry this way.

**Comment by abrichr** [comment]
Thanks @0dm ! Can you please add documentation changes? 🙏

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+readme = ["README.md", "LICENSE"]
+
+repository = "https://github.com/mldsai/openadapt"
+homepage = "https://openadapt.ai/"
+
+
+[tool.poetry.dependencies]
+python = "3.10.x"
+atomacos = {git = "https://github.com/abrichr/atomacos.git", markers = "sys_platform == 'darwin'"}
+pynput = {git = "https://github.com/abrichr/pynput.git"}
```
Comment (abrichr) [inline pyproject.toml:15]: Any way to specify these as editable? Possibly relevant: https://stackoverflow.com/a/69711730/95989

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+nicegui = "1.2.16"
+spacy = "^3.5.3"
+fuzzywuzzy = "0.18.0"
+segment-anything = "^1.0"
+torchvision = "^0.15.2"
+sumy = "0.11.0"
+nltk = "3.8.1"
+
+[tool.poetry.dependencies.en_core_web_trf]
+url = "https://github.com/explosion/spacy-models/releases/download/en_core_web_trf-3.5.0/en_core_web_trf-3.5.0.tar.gz"
```
Comment (abrichr) [inline pyproject.toml:59]: Nice

**Pull request merged**